.github/workflows/node.js.yml

```diff
@@ -10,7 +10,40 @@ on:
         branches: [master, feature/*, mynah-dev]
 
 jobs:
+    lint-commits:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+              with:
+                  fetch-depth: 20
+            - uses: actions/setup-node@v4
+              with:
+                  node-version: '20'
+            - name: Check PR title and commit message
+              run: |
+                  node "$GITHUB_WORKSPACE/.github/workflows/lintcommit.js"
+
+    lint:
+        needs: lint-commits
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                node-version: [16.x]
+                vscode-version: [stable]
+        env:
+            NODE_OPTIONS: '--max-old-space-size=8192'
+        steps:
+            - uses: actions/checkout@v4
+            - name: Use Node.js ${{ matrix.node-version }}
+              uses: actions/setup-node@v4
+              with:
+                  node-version: ${{ matrix.node-version }}
+            - run: npm ci
+            - run: npm run testCompile
+            - run: npm run lint
+
     macos:
+        needs: lint-commits
```

Comment:
The lint-commits job takes <5s. By gating all the jobs on the lint-commits job, it calls out any problems quickly so that the PR title is fixed before proceeding.